docs/Killswitches.md

```diff
@@ -0,0 +1,55 @@
+# Kill Switches
+
+EDMarketConnector implements a Kill Switch system that allows us to disable features based on a version mask. Meaning that we can stop major bugs from affecting the services we support, at the expense of disabling that support.
+
+## Format
+
+Killswitches are stored in a JSON file that is queried by EDMC on startup. The format is as follows:
+
+|             Key |   Type   | Description                                                   |
+| --------------: | :------: | :------------------------------------------------------------ |
+|       `version` |  `int`   | the version of the Kill Switch JSON file, always 1            |
+|  `last_updated` | `string` | When last the kill switches were updated (for human use only) |
+| `kill_switches` | `array`  | The kill switches this file contains (expanded below)         |
+
+The `kill_switches` array contains kill switch objects. Each contains two fields:
+
+|       Key |        Type        | Description                                                             |
+| --------: | :----------------: | :---------------------------------------------------------------------- |
+| `version` | `semantic version` | The version of EDMC these kill switches apply to (Must be valid semver) |
+|   `kills` |  `Dict[str, str]`  | The different keys to disable, and the reason for the disable           |
+An example follows:
+
+```json
+{
+    "version": 1,
+    "last_updated": "19 October 2020",
+    "kill_switches": [
+        {
+            "version": "1.0.0",
+            "kills": {
+                "plugins.eddn.send": "some reason"
+            }
+        }
+    ]
+}
+```
+
+### Versions
+
+Versions are checked using equality checks on `semantic_version.Version` instances. Meaning that **all** fields are checked (ie, Major, Minor, Patch, Prerelease, and Build).
```

Comment:
IIRC there's support in semantic_version for things like a <= comparison.  We *might* discover that something affects "every version before we fixed it" and wouldn't want to therefor list every version (since killswitch support was added) up to the last broken one.  It would be cleaner to be able to specify a version as "<= 4.2.6" if we fixed it in 4.2.7.

Reply:
Yeah I was poking at it a little, https://python-semanticversion.readthedocs.io/en/latest/reference.html#semantic_version.SimpleSpec looks to be what would work for it I think?

Reply:
Yup, exactly what I had in mind.  I'll leave you to figure out the most succinct way of making use of it :) .